pkg/varlog/operations.go

```diff
@@ -147,3 +148,61 @@ func (v *logImpl) logStreamReplicaMetadata(ctx context.Context, tpID types.Topic
 	}
 	return snpb.LogStreamReplicaMetadataDescriptor{}, err
 }
+
+func (v *logImpl) peekLogStream(ctx context.Context, tpid types.TopicID, lsid types.LogStreamID) (first varlogpb.LogSequenceNumber, last varlogpb.LogSequenceNumber, err error) {
+	replicas, ok := v.replicasRetriever.Retrieve(tpid, lsid)
+	if !ok {
+		err = errNoLogStream
+		return
+	}
+
+	var (
+		errs  = make([]error, len(replicas))
+		wg    sync.WaitGroup
+		mu    sync.Mutex
+		found bool
+	)
+	for idx := range replicas {
+		wg.Add(1)
+		go func(idx int) {
+			defer wg.Done()
+
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			client, erri := v.logCLManager.GetOrConnect(ctx, replicas[idx].StorageNodeID, replicas[idx].Address)
+			if erri != nil {
+				errs[idx] = erri
+				return
+			}
+			lsrmd, erri := client.LogStreamReplicaMetadata(ctx, tpid, lsid)
+			if erri != nil {
+				errs[idx] = erri
+				return
+			}
+			switch lsrmd.Status {
+			case varlogpb.LogStreamStatusRunning, varlogpb.LogStreamStatusSealed:
+				mu.Lock()
+				defer mu.Unlock()
+				if first.LLSN < lsrmd.LocalLowWatermark.LLSN {
```

Comment:
Does it means highest LocalLowWatermark?

Reply:
Yes, it is. Since the trim operation removes from the front of log entries, PeekLogStream takes the largest one.